### PR TITLE
Helper: Various Updates

### PR DESCRIPTION
**Pull request by kovshenin**
Various updates to the Helper, including:

* Better support for lifetime subscriptions
* Fixed core plugin information modal for most WooCommerce.com extensions
* Added an number icon indicating number of available updates to WooCommerce extensions
* New subscription filters
* Add "shared by" label for shared subscriptions

### CODE REVIEW

File: includes/admin/helper/class-wc-helper-plugin-info.php [hunk 1]
```diff
@@ -36,25 +36,27 @@ public static function plugins_api( $response, $action, $args ) {
...
-
-		// Look through local Woo plugins by slugs.
-		foreach ( WC_Helper::get_local_woo_plugins() as $plugin ) {
-			if ( dirname( $plugin['_filename'] ) === $args->slug ) {
-				$plugin['_slug'] = $args->slug;
-				$found_plugin = $plugin;
-				break;
-			}
+		// Only for slugs that start with woo-
+		if ( 0 !== strpos( $args->slug, 'woo-' ) ) {
```
Comment (mikejolley) [inline includes/admin/helper/class-wc-helper-plugin-info.php]: If this unique enough? woo- seems generic, and wc- would be worse. Maybe something specific to ourselves like `woocommerce-com` which infers it's origin?

File: includes/admin/helper/class-wc-helper-plugin-info.php [hunk 2]
```diff
@@ -36,25 +36,27 @@ public static function plugins_api( $response, $action, $args ) {
...
+			return $response;
 		}
 
-		if ( ! $found_plugin ) {
+		$clean_slug = preg_replace( '#^woo-#', '', $args->slug );
```
Comment (mikejolley) [inline includes/admin/helper/class-wc-helper-plugin-info.php]: If you use a more unique slug, str_replace can be used here instead for speed.

File: includes/admin/helper/class-wc-helper-updater.php [hunk 3]
```diff
@@ -230,10 +231,82 @@ private static function _has_active_subscription( $product_id ) {
...
+		// Don't fetch any new data since this function in high-frequency.
+		if ( ! get_transient( '_woocommerce_helper_subscriptions' ) ) {
+			return 0;
+		}
+
+		if ( ! get_transient( '_woocommerce_helper_updates' ) ) {
+			return 0;
+		}
+
+		$update_data = self::get_update_data();
```
Comment (mikejolley) [inline includes/admin/helper/class-wc-helper-updater.php]: Why doesn't this case set the `_woocommerce_helper_updates_count` transient?

File: includes/admin/views/html-admin-page-addons.php [hunk 4]
```diff
@@ -13,7 +13,17 @@
 <div class="wrap woocommerce wc_addons_wrap">
 	<nav class="nav-tab-wrapper woo-nav-tab-wrapper">
 		<a href="<?php echo esc_url( admin_url( 'admin.php?page=wc-addons' ) ); ?>" class="nav-tab nav-tab-active"><?php _e( 'Browse Extensions', 'woocommerce' ); ?></a>
-		<a href="<?php echo esc_url( admin_url( 'admin.php?page=wc-addons&section=helper' ) ); ?>" class="nav-tab"><?php _e( 'WooCommerce.com Subscriptions', 'woocommerce' ); ?></a>
+
+		<?php
+			$count = WC_Helper_Updater::get_updates_count();
+			$count_html = sprintf( '<span class="update-plugins count-%d"><span class="update-count">(%d)</span></span>', $count, number_format_i18n( $count ) );
```
Comment (mikejolley) [inline includes/admin/views/html-admin-page-addons.php]: This code is repeated 3 times; make a helper method in `WC_Helper_Updater`?

**Review by mikejolley — CHANGES_REQUESTED** [review]
Few feedback items @kovshenin. thanks!

File: includes/admin/helper/class-wc-helper-plugin-info.php (hunk 1 above)
Comment (kovshenin) [reply includes/admin/helper/class-wc-helper-plugin-info.php]: It doesn't require uniqueness. If we come across a plugin with the slug "woo-something-else" which is not on WooCommerce.com, nothing really happens. But I did change it to what you suggested in a59ce3b6b13359ab74f6bfee877f5c10d3795529.

File: includes/admin/helper/class-wc-helper-updater.php (hunk 3 above)
Comment (kovshenin) [reply includes/admin/helper/class-wc-helper-updater.php]: Good catch!

File: includes/admin/views/html-admin-page-addons.php (hunk 4 above)
Comment (kovshenin) [reply includes/admin/views/html-admin-page-addons.php]: 👍

**Comment by belcherj** [comment]
I have made a commit here by making the notification bubble more general.  The css was not applying because the css was not being loaded when not on the helper screen which makes a lot of sense.

Please let me know if you need me to change implementation.

**Pull request merged**